ssl/t1_lib.c

```diff
@@ -1418,6 +1418,7 @@ unsigned char *ssl_add_clienthello_tlsext(SSL *s, unsigned char *buf,
         s2n(s->alpn_client_proto_list_len, ret);
         memcpy(ret, s->alpn_client_proto_list, s->alpn_client_proto_list_len);
         ret += s->alpn_client_proto_list_len;
+        s->s3->alpn_sent = 1;
```

Comment:
Can you explain the `if` block? Is this one of the weird renego checks? I think this deserves a comment.